test/common/deployment_types.go

```diff
@@ -266,39 +294,113 @@ func TestDeploymentExpectedReplicas(t TestingTB, ctx *TestingContext) {
 				continue
 			}
 
-			// Verify that the expected replicas are also available, means they are up and running and consumable by users
-			if deployment.Status.AvailableReplicas < product.ExpectedReplicas {
-				t.Errorf("Deployment %s in namespace %s doesn't match the number of expected available replicas. Available Replicas: %v / Expected Replicas: %v",
-					product.Name,
-					namespace.Name,
-					deployment.Status.AvailableReplicas,
-					product.ExpectedReplicas,
-				)
-				continue
+			if rhmi.Spec.Type == string(integreatlyv1alpha1.InstallationTypeManagedApi) {
+				if product.Name == "ratelimit" {
+					pods := &v13.PodList{}
+					selector, _ := labels.Parse("app=ratelimit")
+					listOpts := []k8sclient.ListOption{
+						k8sclient.InNamespace(Marin3rProductNamespace),
+						k8sclient.MatchingLabelsSelector{
+							Selector: selector,
+						},
+					}
+					err = ctx.Client.List(context.TODO(), pods, listOpts...)
+					if err != nil {
+						t.Fatalf("failed to get pods for Ratelimit: %v", err)
+					}
+					checkPods(t, pods, product, namespace, deployment)
+
+				} else if product.Name == "apicast-production" {
+					pods := &v13.PodList{}
+					selector, _ := labels.Parse("deploymentconfig=apicast-production")
+					listOpts := []k8sclient.ListOption{
+						k8sclient.InNamespace(ThreeScaleProductNamespace),
+						k8sclient.MatchingLabelsSelector{
+							Selector: selector,
+						},
+					}
+					err = ctx.Client.List(context.TODO(), pods, listOpts...)
+					if err != nil {
+						t.Fatalf("failed to get backend listener pods for 3scale: %v", err)
+					}
+					checkPods(t, pods, product, namespace, deployment)
+
+				} else if product.Name == "backend-listener" {
+					pods := &v13.PodList{}
+					selector, _ := labels.Parse("deploymentConfig=backend-listener")
+					listOpts := []k8sclient.ListOption{
+						k8sclient.InNamespace(ThreeScaleProductNamespace),
+						k8sclient.MatchingLabelsSelector{
+							Selector: selector,
+						},
+					}
+					err = ctx.Client.List(context.TODO(), pods, listOpts...)
+					if err != nil {
+						t.Fatalf("failed to get backend listener pods for 3scale: %v", err)
+					}
+					checkPods(t, pods, product, namespace, deployment)
+
+				} else if product.Name == "backend-worker" {
+					pods := &v13.PodList{}
+					selector, _ := labels.Parse("deploymentconfig=backend-worker")
+					listOpts := []k8sclient.ListOption{
+						k8sclient.InNamespace(ThreeScaleProductNamespace),
+						k8sclient.MatchingLabelsSelector{
+							Selector: selector,
+						},
+					}
+					err = ctx.Client.List(context.TODO(), pods, listOpts...)
+					if err != nil {
+						t.Fatalf("failed to get backend listener pods for 3scale: %v", err)
+					}
+					checkPods(t, pods, product, namespace, deployment)
```

Comment:
Bit confused on why there's these blocks for checking 3scale deployment configs here ? :thinking: 
This specific test `TestDeploymentExpectedReplicas` is for `deployments` and I don't think it would ever get 3scale deployment configs in this test 
* https://github.com/integr8ly/integreatly-operator/blob/3ae07fc47fd73ee86a7e18c86054b8d2f40c709b/test/common/deployment_types.go#L261
* https://github.com/integr8ly/integreatly-operator/blob/3ae07fc47fd73ee86a7e18c86054b8d2f40c709b/test/common/deployment_types.go#L385-L405
* https://github.com/integr8ly/integreatly-operator/blob/3ae07fc47fd73ee86a7e18c86054b8d2f40c709b/test/common/deployment_types.go#L24-L45

Should also fail here since deployment resource would not exist for 3scale deployment configs :thinking: 
* https://github.com/integr8ly/integreatly-operator/blob/3ae07fc47fd73ee86a7e18c86054b8d2f40c709b/test/common/deployment_types.go#L280-L285

Were you meaning to put these code blocks into the `TestDeploymentConfigExpectedReplicas` test ? :thinking: 
* https://github.com/integr8ly/integreatly-operator/blob/3ae07fc47fd73ee86a7e18c86054b8d2f40c709b/test/common/deployment_types.go#L407-L451

Reply:
Now I'm wondering how I had those passing. Well spotted. Moved 3scale checks to the `TestDeploymentConfigExpectedReplicas`.